datafusion/expr-common/src/signature.rs

```diff
@@ -213,9 +221,8 @@ pub enum TypeSignatureClass {
     Interval,
     Duration,
     Native(LogicalTypeRef),
-    // TODO:
-    // Numeric
-    // Integer
+    Numeric(LogicalTypeRef),
+    Integer(LogicalTypeRef),
```

Comment:
@shehabgamin I found that we might not need `LogicalTypeRef`.
This is designed to accept all the Integer, so if the given type is integer, we keep it as it is.

If we want specific integer type, then we should use Native instead. Does this makes sense to you?

Numeric is the same